jwt-bearer-auth-yesod/src/Web/Auth/Bearer/JWT/Yesod/Types.hs

```diff
@@ -0,0 +1,75 @@
+-- |
+-- This module provides JWT Bearer authentication types for Yesod applications.
+module Web.Auth.Bearer.JWT.Yesod.Types
+  ( JWTBearerAuthSettings (..)
+  , HasConfiguredKeyStore (..)
+  , ConfiguredStore (..)
+  , settingsExpectedAudience
+  , settingsTokenServerUrl
+  , settingsRefreshDelayMicros
+  ) where
+
+import Prelude
+
+import Control.Lens
+import Crypto.JOSE (JWK)
+import Web.Auth.Bearer.JWT (TokenServerUrl)
+import Web.Auth.Bearer.JWT.Cache (JWKCache)
+
+data ConfiguredStore a = ConfiguredStore
+  { settings :: JWTBearerAuthSettings a
+  , jwkStore :: a
+  }
+
+-- deriving stock (Eq, Show)
+
+-- | Configuration settings for JWT Bearer authentication in Yesod
+data JWTBearerAuthSettings storeType where
```

Comment:
So `storeType` is just a phantom type variable? My understanding is those are useful for tagging distinct types in order to avoid mixing them up. I don't think this library is likely to have multiple store values at different types ever in use at once. Therefore, it feels to me like this is adding cognitive overhead without any real benefit.

Am I missing something?

If not, I think simplifying this type would have cascading simplifications throughout (can we drop the second type variable out of `HasConfiguredStore` too?) and would be worth exploring.

Reply:
Not a phantom type parameter, it's fixed to a concrete store type by each constructor